presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PickLayout.java

```diff
@@ -70,7 +68,7 @@ protected PlanNode visitPlan(PlanNode node, RewriteContext<Void> context)
         @Override
         public PlanNode visitFilter(FilterNode node, RewriteContext<Void> context)
         {
-            if (node.getSource() instanceof TableScanNode && !((TableScanNode) node.getSource()).getLayout().isPresent()) {
+            if (node.getSource() instanceof TableScanNode) {
```

Comment:
Why do we need this change?

Reply:
>  I think it desired that legacy and no legacy rules have the same logic.

I disagree. It is usually desired that _during_ migration we migrate legacy logic into a rule and only then change the rule. 

However, there is no point in keeping them inline. Also, to me, `PickTableLayout` didn't _really_ exist before this PR and we may make it clear here. I mean -- the `PickLayout`, `PickTableLayout` pair was used in tests _only_. So in this PR you can simply remove the old `PickLayout` (and please do so!) without much consideration, since it was never used in production. (or, to be precise -- it was always fired in production, but it always realized layouts are already picked by `AddExchanges`, so there was never anything to do).

Reply:
Ok. I removed these changes. But let leave this class for a while. There is a risk that `PickTableLayout` could loop once again, then disabling itartive optimizer could save someone's life.